src/utils/aotf.js

```diff
@@ -517,20 +517,12 @@ export function getMutationExtendedDesc (text) {
  * @param {IntrospectionInputType[]} types - Types as returned by introspection query.
  */
 export function processArguments (mutation, types) {
-  let pointer = null
-  let multiple = null
-  let required = null
-  let cylcObject = null
-  let cylcType = null
   for (const arg of mutation.args) {
-    pointer = arg.type
-    multiple = false
-    required = false
-    cylcObject = null
-    cylcType = null
-    if (pointer?.kind === 'NON_NULL') {
-      required = true
-    }
+    let pointer = arg.type
+    let multiple = false
+    let cylcObject = null
+    let cylcType = null
+    const required = arg.type?.kind === 'NON_NULL'
```

Comment:
Jslint used to tell us not to declare variables within loops like this. More performant to declare once and reuse?

Reply:
I can't find much info on that and perhaps that dated back to when `var` was used to declare variables?

I prefer this way because a) it reduces duplication and makes the code more readable, b) it is less likely to lead to the value from one iteration accidentally leaking into the next iteration - the scope is only within a single iteration.

Reply:
> dated back to when var was used to declare variables?

That's probably it. Using `var` within a loop would re-declare the variable over and over.

Reply:
Actually the opposite I think `var` declarations were hoisted 😬